programs/bubblegum/program/src/lib.rs

```diff
@@ -1348,6 +1595,110 @@ pub mod bubblegum {
         )
     }
 
+    pub fn update_metadata<'info, 'a>(
+        ctx: Context<'_, '_, 'a, 'info, UpdateMetadata<'info>>,
+        root: [u8; 32],
+        old_metadata: Option<MetadataArgs>,
+        new_name: Option<String>,
+        new_symbol: Option<String>,
+        new_uri: Option<String>,
+        new_creators: Option<Vec<Creator>>,
+        new_seller_fee_basis_points: Option<u16>,
+        new_primary_sale_happened: Option<bool>,
+        new_is_mutable: Option<bool>,
+        nonce: u64,
+        index: u32,
```

Comment:
Nit: This is more of a question than a definite assertion, but should the parameter order be?:
```
root: [u8; 32],
nonce: u64,
index: u32,
old_metadata: Option<MetadataArgs>,
new_name: Option<String>,
new_symbol: Option<String>,
new_uri: Option<String>,
...
```
This seems like it might be more consistent with other instructions that put the `MetadataArgs` last?

Although maybe you were considering all the `MetadataArgs` as a replacement for `data_hash` and `creator_hash`?  For me it is currently making sense to put all the `MetadataArgs` at the end.

Reply:
Yeah, originally I was following a similar model to the `redeem` function:

```
pub fn redeem<'info>(
        ctx: Context<'_, '_, '_, 'info, Redeem<'info>>,
        root: [u8; 32],
        data_hash: [u8; 32],
        creator_hash: [u8; 32],
        nonce: u64,
        index: u32,
    )
```

But replacing `data_hash` and `creator_hash` with the `MetadataArgs` params. But, I think the order you suggested makes the most sense given the large number of `MetadataArgs` params. That way the order of params is roughly: `stuff identifying the state of the tree and which leaf you want to update`, `stuff indicating how you want to update it`